lnbits/core/views/api.py

```diff
@@ -387,16 +408,13 @@ async def api_payments_sse(
 async def api_payment(payment_hash, X_Api_Key: Optional[str] = Header(None)):
     # We use X_Api_Key here because we want this call to work with and without keys
     # If a valid key is given, we also return the field "details", otherwise not
-    wallet = None
-    try:
-        if X_Api_Key.extra:
-            logger.warning("No key")
-    except:
-        wallet = await get_wallet_for_key(X_Api_Key)
+    wallet = await get_wallet_for_key(X_Api_Key) if type(X_Api_Key) == str else None
```

Comment:
Let's hope this is safe. But I've tested it multiple times now, it should be ok.

Reply:
i added tests for that,  in this commit, right? the ones you gave me.